dcos/rpcclient.py

```diff
@@ -145,7 +144,8 @@ def http_req(self, method_fn, path, *args, **kwargs):
                 request_method=e.response.request.method,
                 request_url=e.response.request.url,
                 json_body=json_body)
-            raise DCOSException(message)
+
+            raise DCOSException(message) from e
```

Comment:
Ah, so I don't use the CLI but the dcos Python client that comes with the CLI `dcos.marathon`. It's used in Shakedown and some exception are pretty useless when they swallow the actual exception. The debug line I added help here. However, I don't it's good practice to catch an exception and throw a new one. IMHO the CLI should get as much information as possible and decide how to present what to the user.

So  it pretty much comes down to shakedown using this package as a kind of Python client for DC/OS. It also adds other methods.